### DISCUSSION TITLE
How to display number of specific records on parent's table column from child's database?

### DISCUSSION THREAD

**Discussion by Hypertechman** [filamentphp/filament#14284] [Help] [answered]
### Package

Table builder

### Package Version

v3.x

### How can we help you?

Hello, Im kind of new in Filament, I have a parent resource called sesi and a child resource called senarai. In the senarai resource, there is a column called status which is a boolean. What I want to do is to have the total of status=1 and status=0 to be displayed on sesi's table column. 

```
//child
//senarai model
public function sesiAsnaf(): BelongsTo
    {
       return $this->belongsTo(sesiAsnaf::class);
    }
```

```
//parent
//sesi model
public function senaraiAsnafs(): HasMany
    {
        return $this->hasMany(senaraiAsnaf::class);
    }
```
![image](https://github.com/user-attachments/assets/d4e7a113-8ad2-4546-956a-d8103e63a7c4)

Above is the sesi table that shows the content of the senarai, however what I want is the total of the 1 and 0. So there should be 2 columns for each.

For extra information Ive followed Modestas's nested resource trait pages guide in the past where when I click on the parent row it will redirect to the child page with the same sesi_id.

**Comment by leandrocfe** [comment, marked as answer]
Maybe customizing it via `state` ?
https://filamentphp.com/docs/3.x/tables/columns/getting-started#calculated-state

**Reply by Hypertechman** [reply]
Ive tried using state but seems like it have problem with the datatype
```
App\Filament\Resources\SesiAsnafResource::App\Filament\Resources\{closure}(): Argument #1 ($record) must be of type
App\Models\SenaraiAsnaf, App\Models\sesiAsnaf given, called in C
\Workspace\firstLaravel\vendor\filament\support\src\Concerns\EvaluatesClosures.php on line 35 
```
The db datatype is bool, changed it to int but still produces the same error. As far as I know, state supposedly can obtain value from other model as I wanted to but it doesnt seems to working even with simple code like
```
//This is sesiAsnaf resource (parent)
use App\Models\SenaraiAsnaf;

Tables\Columns\TextColumn::make('jumlahSudahMenerima')
                ->state(function(SenaraiAsnaf $record) {
                //SenaraiAsnaf is child model
                    return $record->zakatStatus;
                })
```
Does it automatically sum all the zakatStatus value or do i need something like sum() to those value? I also need it to only sum the list where the foreign id = x.

**Comment by Hypertechman** [comment]
I found the way using state as @leandrocfe suggested.
```
Tables\Columns\TextColumn::make('jumlahSudahTerima')
                ->state(function($record, $column) {
                    $name = $column->getName();
                    $record->update([
                        $name => $record->senaraiAsnafs()->sum('zakatStatus')
                    ]);
                    return $record->senaraiAsnafs()->sum('zakatStatus');
                }),
```
I'm not sure if there is prettier way to do this but it got the job done. Since the value is in bool, I use sum() to get total of '1's, and use count() to get total record in the column. Then I use count()-sum() to get number of '0's. It did update both the table and list. Thank you.
